prov/efa/src/efa_cq.c

```diff
@@ -175,8 +179,13 @@ ssize_t efa_cq_readfrom(struct fid_cq *cq_fid, void *buf, size_t count,
 								  wc.ibv_wc.slid,
 								  wc.ibv_wc.src_qp);
 		}
-		cq->read_entry(&wc, i, buf);
-	}
+		cq->read_entry(&wc, i++, buf);
+
+		ret = ibv_next_poll(cq->ibv_cq_ex);
+	} while (i < count);
+
+	if (should_end_poll)
+		ibv_end_poll(cq->ibv_cq_ex);
 
 	ofi_spin_unlock(&cq->lock);
 	return i ? i : ret;
```

Comment:
Maybe outside the scope of this PR, but I do not think this return value is right.

if ret < 0 and ret != -FI_EAGAIN,  we should returned ret here, instead of i.

Reply:
Good question. I'm not sure. I think the only way to verify is to write a test against the error path.